packages/metro-serializer-esbuild/src/index.ts

```diff
@@ -44,15 +44,38 @@ function fixSourceMap(outputPath: string, text: string): string {
   return JSON.stringify({ ...sourcemap, sources });
 }
 
-function getSideEffects(modulePath: string): boolean | undefined {
-  const pkgJson = findPackage(modulePath);
-  if (!pkgJson) {
-    return undefined;
-  }
+const getSideEffects = (() => {
+  const pkgCache: Record<string, boolean | string[]> = {};
+  const getSideEffects = (pkgJson: string) => {
+    if (!(pkgJson in pkgCache)) {
+      const content = fs.readFileSync(pkgJson, { encoding: "utf-8" });
+      const { sideEffects } = JSON.parse(content);
```

Comment:
```suggestion
      const content = readPackage(pkgJson);
      const { sideEffects } = content;
```
From `@rnx-kit/tools-node`